lc-hax/Scripts/Patches/ChatPatch.cs

```diff
@@ -9,8 +9,13 @@ namespace Hax;
 [HarmonyPatch("EnableChat_performed")]
 class EnableChatPatch {
     static void Prefix(ref PlayerControllerB ___localPlayer, ref bool __state) {
-        __state = ___localPlayer.isPlayerDead;
-        ___localPlayer.isPlayerDead = false;
+        if (___localPlayer is null) {
+            __state = false;
+        }
+        else {
+            __state = ___localPlayer.isPlayerDead;
+            ___localPlayer.isPlayerDead = false;
+        }
     }
 
     static void Postfix(ref PlayerControllerB ___localPlayer, bool __state) => ___localPlayer.isPlayerDead = __state;
```

Comment:
I'd prefer if we avoid nesting wherever possible.

```cs
  static void Prefix(ref PlayerControllerB? ___localPlayer, ref bool __state) {
      if (!___localPlayer.IsNotNull(out PlayerControllerB localPlayer)) return;

      __state = localPlayer.isPlayerDead;
      localPlayer.isPlayerDead = false;
  }

  static void Postfix(ref PlayerControllerB? ___localPlayer, bool __state) {
      if (!___localPlayer.IsNotNull(out PlayerControllerB localPlayer)) return;
      localPlayer.isPlayerDead = __state;
  }
```

I think we need to check for `null` in `Postfix` as well since it may also cause an exception.

Reply:
`localPlayer` gets assigned by `EnableChat_performed` itself so it's not necessary in `Postfix`, haven't seen any error since my `Prefix` edit

Reply:
I see. Then just this I guess.

```cs
  static void Prefix(ref PlayerControllerB? ___localPlayer, ref bool __state) {
      if (!___localPlayer.IsNotNull(out PlayerControllerB localPlayer)) return;

      __state = localPlayer.isPlayerDead;
      localPlayer.isPlayerDead = false;
  }
```